src/coreComponents/constitutive/fluid/multifluid/compositional/functions/RachfordRice.hpp

```diff
@@ -98,48 +98,37 @@ struct RachfordRice
     real64 currentError = 1 / epsilon;
 
     // step 2: start the SSI loop
-    real64 funcXMin = 0.0;
+    // Evaluate at the bounds
+    real64 funcXMin = evaluate( kValues, feed, presentComponentIds, xMin );
+    real64 funcXMax = evaluate( kValues, feed, presentComponentIds, xMax );
     real64 funcXMid = 0.0;
-    real64 funcXMax = 0.0;
-    bool recomputeMin = true;
-    bool recomputeMax = true;
+
+    // If the bound values are the same sign then we have a trivial solution
+    if( 0.0 < funcXMin * funcXMax )
+    {
+      gasPhaseMoleFraction = (0.0 < funcXMin) ? 1.0 : 0.0;
+      return gasPhaseMoleFraction;
+    }
+
     integer SSIIteration = 0;
 
     while( ( currentError > SSITolerance ) && ( SSIIteration < maxSSIIterations ) )
     {
       real64 const xMid = 0.5 * ( xMin + xMax );
-      if( recomputeMin )
-      {
-        funcXMin = evaluate( kValues, feed, presentComponentIds, xMin );
-      }
-      if( recomputeMax )
-      {
-        funcXMax = evaluate( kValues, feed, presentComponentIds, xMax );
-      }
       funcXMid = evaluate( kValues, feed, presentComponentIds, xMid );
 
-      if( ( funcXMin < 0 ) && ( funcXMax < 0 ) )
-      {
-        return gasPhaseMoleFraction = 0.0;
-      }
-      else if( ( funcXMin > 1 ) && ( funcXMax > 1 ) )
-      {
-        return gasPhaseMoleFraction = 1.0;
-      }
-      else if( funcXMin * funcXMid < 0.0 )
+      if( 0.0 < funcXMax * funcXMid )
       {
         xMax = xMid;
-        recomputeMax = true;
-        recomputeMin = false;
+        funcXMax = funcXMid;
       }
-      else if( funcXMax * funcXMid < 0.0 )
+      else if( 0.0 < funcXMin * funcXMid )
       {
         xMin = xMid;
-        recomputeMax = false;
-        recomputeMin = true;
+        funcXMin = funcXMid;
       }
 
-      currentError = LvArray::math::min( LvArray::math::abs( funcXMax - funcXMin ),
+      currentError = LvArray::math::min( LvArray::math::abs( funcXMid ),
```

Comment:
We should maybe take a careful look at the `currentError` metric and potential scaling issues here and in the Newton loop below.  I think the absolute error metric is mostly okay here in SSI since everything should be hopefully order 1 near the root, but the error metric in the newton loop below at line 148 should be fixed.  If `newtonValue` approaches zero, the error may appear very large when it is not.  I suggest we use a very similar error measure in both SSI and newton.  `deltaNewton` serves the same purpose as `xMax-xMin`, with an absolute value checking having the meaning of the precision in phase saturation we need, always bounded between 0 and 1.   Maybe we should add a function value check to Newton too.  To be safe, I might also scale the function evaluation by the function evaluated at the midpoint (x=0.5) or some other starting guess, in case the function values have crazy scaling.  That one may benefit from a relative error check versus an absolute error check.  What do you think?

Reply:
On second thought, maybe both the `deltaNewton` and `functionValue` checks should be absolute, with no relative scaling at all.  There may be situations where the initial guess for `functionValue` is near perfect.
```
SSI:  currentError = min( abs( funcXMid ), abs( xMax - xMin ) );
Newt: currentError = min( abs( newtonValue ), abs( deltaNewton ) );
```

Reply:
> 
> ```
> Newt: currentError = min( abs( newtonValue ), abs( deltaNewton ) );
> ```

I would suppose we want
```
Newt: currentError = min( abs( evaluate(newtonValue) ), abs( deltaNewton ) );
```
i.e. the function value at the newtonValue not the newtonValue itself.

Reply:
Whoops, yes, exactly